protocols/kad/src/behaviour.rs

```diff
@@ -1385,6 +1387,15 @@ where
         }
     }
 
+    /// A new peer has been inserted in the routing table but we check if the routing
+    /// table is currently small (less that `K_VALUE` peers are present) and only
+    /// trigger a bootstrap in that case
+    fn trigger_bootstrap_if_routing_table_is_almost_empty(&mut self) {
+        if self.kbuckets().count() < K_VALUE.get() {
```

Comment:
IIUC `self.kbuckets().count()` returns the number of buckets, not to number of peers in the routing table.

The following _should_ work, but it would be good to add a test.
```suggestion
        let local_key = *self.kbuckets.local_key();
        if self.kbuckets.closest_keys(&local_key).count() < K_VALUE.get() {
```

Reply:
Yes of course. I don't know how I've missed that. I did it slightly differently. Tell me what you think.

```rust
self.kbuckets().map(|kbucket| kbucket.num_entries()).sum::<usize>()
```

Reply:
Perfect!